sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py

```diff
@@ -93,13 +94,38 @@ class _Infinity:
 class PartitionKey(dict):
     """Key used to partition a container into logical partitions.
 
-    See https://learn.microsoft.com/azure/cosmos-db/partitioning-overview#choose-partitionkey
-    for information on how to choose partition keys.
-
-    :ivar str path: The path of the partition key
-    :ivar str kind: What kind of partition key is being defined (default: "Hash")
-    :ivar int version: The version of the partition key (default: 2)
-    """
+See https://learn.microsoft.com/azure/cosmos-db/partitioning-overview#choose-partitionkey
+for information on how to choose partition keys.
+
+This constructor supports multiple overloads:
+
+1. **Single Partition Key**:
+   - **Parameters**:
+     - `path` (str): The path of the partition key.
+     - `kind` (Literal["Hash"], optional): The kind of partition key. Defaults to "Hash".
+     - `version` (int, optional): The version of the partition key. Defaults to 2.
+   - **Example**:
+     >>> pk = PartitionKey(path="/id")
+
+2. **Hierarchical Partition Key**:
+   - **Parameters**:
+     - `path` (List[str]): A list of paths representing the partition key, supporting up to three hierarchical levels.
+     - `kind` (Literal["MultiHash"], optional): The kind of partition key. Defaults to "MultiHash".
+     - `version` (int, optional): The version of the partition key. Defaults to 2.
+   - **Example**:
+     >>> pk = PartitionKey(path=["/id", "/category"], kind="MultiHash")
+
+3. **Partition Key Definition**:
+   - **Parameters**:
+     - `partition_key_definition` (Union[PartitionKey, Dict[str, Any]]): A dictionary or `PartitionKey`
+        object defining the partition key, including `path`, `kind`, and `version`.
+   - **Example**:
+     >>> pk = PartitionKey(partition_key_definition={"path": "/id", "kind": "Hash", "version": 1})
```

Comment:
We should default this documentation to version: 2, because we want customers to create the container with version 2 partition key definition.

Reply:
requested changes were made